indispensable-asn1/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -59,38 +55,40 @@ class ObjectIdentifier @Throws(Asn1Exception::class) private constructor(
      * for details.
      * Lazily evaluated.
      */
-    val bytes: ByteArray by if (bytes != null) lazyOf(bytes) else lazy {
-        this.nodes.toOidBytes()
+    val bytes: ByteArray by lazy {
+        bytes ?: nodes!!.toOidBytes()
     }
 
     /**
      * Lazily evaluated list of OID nodes (e.g. `[1, 2, 35, 4654]`)
      */
-    val nodes by if (nodes != null) lazyOf(nodes) else lazy {
-        val (first, second) =
-            if (this.bytes[0] >= 80) {
-                BigInteger.fromUByte(2u) to BigInteger.fromUInt(this.bytes[0].toUByte() - 80u)
-            } else {
-                BigInteger.fromUInt(this.bytes[0].toUByte() / 40u) to BigInteger.fromUInt(this.bytes[0].toUByte() % 40u)
-            }
-        var index = 1
-        val collected = mutableListOf(first, second)
-        while (index < this.bytes.size) {
-            if (this.bytes[index] >= 0) {
-                collected += BigInteger.fromUInt(this.bytes[index].toUInt())
-                index++
-            } else {
-                val currentNode = Buffer() //todo: calculate only index and then operate on a byte view (unsafe wrapped). probably irrelevant since we'll roll our own
-                while (this.bytes[index] < 0) {
-                    currentNode.writeByte(this.bytes[index]) //+= parsed
+    val nodes: List<String> by lazy {
+        if (nodes != null) nodes.map { it.toString() } else {
```

Comment:
would prefer `if (nodes != null) lazy { nodes.map(VarUInt::toString) } else lazy { ...`

Reply:
then then type will change to. `Lazy<List<String>` instead of `List<String>`

Reply:
no, `by if ...`

Reply:
looks cursed to me no matter where i wrap it

<img width="519" alt="image" src="https://github.com/user-attachments/assets/6ed4b6e3-76b1-48b6-b425-535b4d94351d">


<img width="659" alt="image" src="https://github.com/user-attachments/assets/5dde1e0b-5fa6-497e-b7be-02012450beab">

<img width="735" alt="image" src="https://github.com/user-attachments/assets/258cdb8c-812a-42bc-99c9-5e0da4ed0221">

Reply:
can potentially do `by lazy(if (nodes != null) ({ nodes.map(WhateverType::toString) }) else ({ ... })`